src/utilityFunctions.ts

```diff
@@ -70,15 +71,23 @@ function createDictionaryWithVector<T extends keyof ArrayToArrayMap>(
       'values must be an array of signed integers, 32 bit or smaller.',
     );
   }
-  const type = new Dictionary(labelsArrow.type, t, currentDictNumber++, false);
-  const returnval = makeVector({
-    type,
+  const type: Dictionary<Utf8, ArrayToArrayMap[T]> = new Dictionary(
+    labelsArrow.type,
+    t,
+    currentDictNumber++,
+    false,
+  );
+
+  // @ts-expect-error These are correct and unit tested, but
+  // the typing fails for reasons I don't understand.
+  const props: DataProps<Dictionary<Utf8, ArrayToArrayMap[T]>> = {
```

Comment:
No clue why this code started throwing a ts error, but I've added a unit test to confirm it works